app/src/main/java/com/amaze/filemanager/activities/MainActivity.java

```diff
@@ -548,32 +567,37 @@ private void checkForExternalIntent(Intent intent) {
      * Initializes the floating action button to act as to save data from an external intent
      */
     private void initFabToSave(final ArrayList<Uri> uris) {
+        initFabToSaveInternal(v -> {
+            FileUtil.writeUriToStorage(MainActivity.this, uris, getContentResolver(), getCurrentMainFragment().getCurrentPath());
+            Toast.makeText(MainActivity.this, R.string.saving, Toast.LENGTH_LONG).show();
+            finish();
+        });
+    }
+
+    private void initFabToSave(@NonNull final CharSequence textContent) {
+        initFabToSaveInternal(v -> {
+            mainActivityHelper.add(MainActivityHelper.NEW_FILE, (file) -> {
+                try {
+                    FileUtil.writeContentToHybridFile(MainActivity.this, file, textContent);
+                    runOnUiThread(() -> Toast.makeText(MainActivity.this, getString(R.string.shared_content_saved, file.getName(MainActivity.this)), Toast.LENGTH_LONG).show());
```

Comment:
Is `runOnUiThread` necessary?

Reply:
OK, now I got an explanation. Without `runOnUiThread()` Robolectric tests passed (one-by-one only though) but with exception

```
Exception in thread "AsyncTask #1" java.lang.RuntimeException: An error occured while executing doInBackground()
	at org.robolectric.shadows.ShadowAsyncTask$1.done(ShadowAsyncTask.java:56)
	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:384)
	at java.util.concurrent.FutureTask.setException(FutureTask.java:251)
	at java.util.concurrent.FutureTask.run(FutureTask.java:271)
	at org.robolectric.shadows.ShadowAsyncTask$3.run(ShadowAsyncTask.java:108)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.RuntimeException: Can't toast on a thread that has not called Looper.prepare()
	at android.widget.Toast$TN.__constructor__(Toast.java:390)
	at android.widget.Toast$TN.<init>(Toast.java)
	at android.widget.Toast.__constructor__(Toast.java:114)
	at android.widget.Toast.<init>(Toast.java)
	at org.robolectric.shadows.ShadowToast.makeText(ShadowToast.java:38)
	at android.widget.Toast.makeText(Toast.java)
	at com.amaze.filemanager.activities.MainActivity.lambda$null$3(MainActivity.java:584)
	at com.amaze.filemanager.utils.MainActivityHelper$3.done(MainActivityHelper.java:459)
	at com.amaze.filemanager.filesystem.Operations$2.doInBackground(Operations.java:321)
	at com.amaze.filemanager.filesystem.Operations$2.doInBackground(Operations.java:198)
	at org.robolectric.util.ReflectionHelpers$7.run(ReflectionHelpers.java:216)
	at org.robolectric.util.ReflectionHelpers.traverseClassHierarchy(ReflectionHelpers.java:360)
	at org.robolectric.util.ReflectionHelpers.callInstanceMethod(ReflectionHelpers.java:210)
	at org.robolectric.shadows.ShadowAsyncTaskBridge.doInBackground(ShadowAsyncTaskBridge.java:20)
	at org.robolectric.shadows.ShadowAsyncTask$BackgroundWorker.call(ShadowAsyncTask.java:146)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	... 4 more
```

Reply:
Was wondering where that was coming from... The cost of the call if you already are in UI is negligible, so I have no problem with it.